mycpp/gc_dict.h

```diff
@@ -37,7 +37,13 @@ List<T>* ListFromDictSlab(Slab<int>* index, Slab<T>* slab, int n) {
 template <class K, class V>
 class Dict {
  public:
-  Dict() : GC_CLASS_FIXED(header_, field_mask(), sizeof(Dict)) {
+  Dict()
+      : GC_CLASS_FIXED(header_, field_mask(), sizeof(Dict)),
+        len_(),
+        capacity_(),
+        entry_(),
+        keys_(),
+        values_() {
     assert(len_ == 0);
```

Comment:
OK I think it probably makes sense to switch to this style ... The implicit zero initialization was left over from the Cheney heap, where you allocate within a big block that is zero'd.

Right now `MarkSweepHeap` actually use calloc() which zeroes the buffer, but I think we could switch to `malloc()` and rely on C++ initialization.  The only cases I know where we need custom initialization are the flexible arrays `data_[1]`  in `Str` and `Slab`.